src/screens/WalletConnect/CallRequest/TransactionRequestContent.js

```diff
@@ -170,11 +170,31 @@ const useTransactionFee = (request: WalletConnectCallRequest) => {
   };
   const hasNotEnoughGas = !isEnoughBalanceForTransactionFee(walletBalances, balanceCheckTransaction, chain);
 
-  const { estimateCallRequestTransaction } = useWalletConnect();
+  const { estimateCallRequestTransaction, callRequests } = useWalletConnect();
 
-  React.useEffect(() => estimateCallRequestTransaction(request), [request, estimateCallRequestTransaction]);
+  React.useEffect(() => {
+    // perform additional check to avoid estimate on request dismiss
+    const requestExists = callRequests.some(({ callId }) => callId === request.callId);
+    if (!requestExists) return;
 
-  return { fee, gasSymbol, gasAddress, hasNotEnoughGas, isEstimating, estimationErrorMessage };
+    estimateCallRequestTransaction(request);
+  }, [request, estimateCallRequestTransaction, callRequests]);
```

Comment:
Looks like we could improve the things a bit by applying `useCallback` to `estimateCallRequestTransaction`. While the whole `useWalletConnect` result is memoized, the particular callback should stay the same since it's only calling action method. Same for `approveCallRequest` and `rejectCallRequest`

Reply:
Added for all calls.